### PR TITLE
Support alert server HA

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

Right now, we can only support start single alert server, and need to write the alert server ip in worker's application.yml. 

This will have some problems:

- Once the single alert server crash the alert cannot use.
- Once the alert server ip changed, we need to update the worker's application.yml and restart the worker.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistryClient.java [hunk 1]
```diff
@@ -98,6 +103,15 @@ private void registry() {
         log.info("Worker node: {} registry finished", workerConfig.getWorkerAddress());
     }
 
+    public Host getAlertServerAddress() {
```
Comment (kezhenxu94) [inline dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistryClient.java]: What about returning an `Optional<Host>` here?

File: dolphinscheduler-registry/dolphinscheduler-registry-api/src/main/java/org/apache/dolphinscheduler/registry/api/enums/RegistryNodeType.java [hunk 2]
```diff
@@ -15,32 +15,33 @@
...
 
-    public void setPort(final int port) {
-        this.port = port;
+    public String getName() {
+        return name;
     }
 
-    public int getWaitTimeout() {
-        return waitTimeout;
+    public String getRegistryPath() {
```
Comment (kezhenxu94) [inline dolphinscheduler-registry/dolphinscheduler-registry-api/src/main/java/org/apache/dolphinscheduler/registry/api/enums/RegistryNodeType.java]: Let's use Lombok to simplify these methods

File: dolphinscheduler-standalone-server/src/main/resources/application.yaml [hunk 3]
```diff
@@ -164,22 +164,17 @@ worker:
...
-  max-cpu-load-avg: -1
+  max-cpu-load-avg: 80
   # worker reserved memory, only lower than system available memory, worker server can be dispatched tasks. default value 0.3, the unit is G
-  reserved-memory: 0.3
-  # default worker groups separated by comma, like 'worker.groups=default,test'
-  groups:
-    - default
-  # alert server listen host
-  alert-listen-host: localhost
-  alert-listen-port: 50052
```
Comment (kezhenxu94) [inline dolphinscheduler-standalone-server/src/main/resources/application.yaml:175]: Can you review these changed configurations in the Helm Chart and see whether we have these there, if so remove them as well?

**Review by kezhenxu94 — COMMENTED** [review]
The idea looks good!

File: dolphinscheduler-standalone-server/src/main/resources/application.yaml (hunk 3 above)
Comment (ruanwenjun) [reply dolphinscheduler-standalone-server/src/main/resources/application.yaml:175]: I find these exist in `kubernetes` and `terraform`, and remove these, please help to check again.

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistryClient.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistryClient.java:106]: Done

File: dolphinscheduler-registry/dolphinscheduler-registry-api/src/main/java/org/apache/dolphinscheduler/registry/api/enums/RegistryNodeType.java (hunk 2 above)
Comment (ruanwenjun) [reply dolphinscheduler-registry/dolphinscheduler-registry-api/src/main/java/org/apache/dolphinscheduler/registry/api/enums/RegistryNodeType.java]: Done

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistryClient.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistryClient.java]: Done

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**